freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/stop/StackStopRestartAction.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.freeipa.flow.stack.stop;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Payload;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.flow.core.restart.DefaultRestartAction;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.DetailedStackStatus;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.stack.StackService;
+import com.sequenceiq.freeipa.service.stack.StackUpdater;
+
+@Component("StackStopRestartAction")
+public class StackStopRestartAction extends DefaultRestartAction {
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private StackUpdater stackUpdater;
+
+    @Override
+    public void restart(FlowParameters flowParameters, String flowChainId, String event, Object payload) {
+        Payload stackPayload = (Payload) payload;
+        Stack stack = stackService.getByIdWithListsInTransaction(stackPayload.getResourceId());
```

Comment:
This fetches a stack entity with all its dependencies by id, then the rest of this method only uses the id, which was already present.